sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -839,6 +856,36 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
     if (isUpdate) {
       webResource.headers.set("If-Match", "*");
     }
+
+    const forwardTo = (entityFields as InternalQueueOptions | InternalSubscriptionOptions)
```

Comment:
We are repeating the `(entityFields as InternalQueueOptions | InternalSubscriptionOptions)` in 4 places in this function. The call to `getToken()` is also made twice. I would suggest a simplification as below:

```javascript
    const queueOrSubscriptionFields = entityFields as InternalQueueOptions | InternalSubscriptionOptions;
    if (
      queueOrSubscriptionFields.ForwardTo ||
      queueOrSubscriptionFields.ForwardDeadLetteredMessagesTo
    ) {
      const token = (await this.sasTokenProvider.getToken(this.endpoint)).token;
      const urlPrefix = `https://${this.endpoint}/`;
      if (queueOrSubscriptionFields.ForwardTo) {
        webResource.headers.set("ServiceBusSupplementaryAuthorization", token);
        queueOrSubscriptionFields.ForwardTo = urlPrefix.concat(queueOrSubscriptionFields.ForwardTo);
      }
      if (queueOrSubscriptionFields.ForwardDeadLetteredMessagesTo) {
        webResource.headers.set("ServiceBusDlqSupplementaryAuthorization", token);
        queueOrSubscriptionFields.ForwardDeadLetteredMessagesTo = urlPrefix.concat(
          queueOrSubscriptionFields.ForwardDeadLetteredMessagesTo
        );
      }
    }
```

Reply:
Updated